From f6a25876d1035a34bfafb4bde590c260d5b49980 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Sat, 2 Aug 2025 04:36:06 +0200
Subject: [PATCH 17/162] block: Clean up elevator_set_default

In case of a multi-queue device, the code pointlessly loaded the
default elevator just to drop it again.
---
 block/elevator.c | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/block/elevator.c b/block/elevator.c
index ef808b3ad595..046189e0b03d 100644
--- a/block/elevator.c
+++ b/block/elevator.c
@@ -746,21 +746,22 @@ void elevator_set_default(struct request_queue *q)
 	/*
 	 * For single queue devices, default to using mq-deadline. If we
 	 * have multiple queues or mq-deadline is not available, default
 	 * to "none".
 	 */
+	if (q->nr_hw_queues != 1 && !blk_mq_is_shared_tags(q->tag_set->flags))
+		return;
+
 	ctx.type = elevator_find_get(ctx.name);
 	if (!ctx.type)
 		return;
 
-	if ((q->nr_hw_queues == 1 ||
-			blk_mq_is_shared_tags(q->tag_set->flags))) {
-		err = elevator_change(q, &ctx);
-		if (err < 0)
-			pr_warn("\"%s\" elevator initialization, failed %d, falling back to \"none\"\n",
-					ctx.name, err);
-	}
+	err = elevator_change(q, &ctx);
+	if (err < 0)
+		pr_warn("\"%s\" elevator initialization, failed %d, falling back to \"none\"\n",
+				ctx.name, err);
+
 	elevator_put(ctx.type);
 }
 
 void elevator_set_none(struct request_queue *q)
 {
-- 
2.52.0

